### ISSUE TITLE
Notifications of new messages are not getting displayed on chatwoot Widget.

### ISSUE THREAD

**Issue by hasnat-venturenox** [chatwoot/chatwoot#9967]
### Describe the bug

we are using self hosted instance of chatwoot. The issue we are facing is that whenever our agent responed to the chat the notifications of new messages did not displayed over the chat widget untill I refresh the site. 

we are authentication user with setuser call by providing email, name , and hash. Here is the code that we are using .

### To Reproduce

When a user logs in to our platform, the widget does not notify about any replies that were received from our support while the user was away. But if I refresh the website, I can see the red dot and the reply preview.

The onload function is running when the user logs in and when the page refreshes. The data that's being passed to the setUser function is same in both cases. Here's the chatwoot script we're using for reference.

```
useEffect(() => {
        const loadChatwootScript = () => {
            const g = document.createElement('script');
            const s = document.getElementsByTagName('script')[0];
            g.src = `${baseUrl}/packs/js/sdk.js`;
            g.defer = true;
            g.async = true;
            s.parentNode.insertBefore(g, s);
            g.onload = async () => {
                window.chatwootSDK.run({
                    websiteToken: websiteToken,
                    baseUrl: baseUrl
                });
                if (auth?.data?.access_token && userProfileData ) {
                    // const accessToken = auth.data?.access_token;
                    const result = dispatch(GenerateHash());
                    const hash = result.payload[1].hash;
                    console.log("hash",hash)
                    console.log("email",auth?.data?.email,)
                    console.log("name",USER_NAME)
                    if (hash)
                    {
                        window.$chatwoot.setUser(auth?.data?.user_id.toString(), {
                            email: auth?.data?.email,
                            name: USER_NAME,
                            identifier_hash: hash,
                        });
                    }
            }
            };
        };
        if (typeof document !== 'undefined') {
            loadChatwootScript();
        }
    }, [baseUrl, websiteToken, auth.data.access_token, userProfileData, data]);
```

### Expected behavior

The messages should get displayed in realtime when the agent send messages. 

### Environment

Docker

### Cloud Provider

DigitalOcean

### Platform

Browser

**Comment by pranavrajs** [comment]
@hasnat-venturenox Is this happening every time or randomly?

**Comment by hasnat-venturenox** [comment]
Yes this happens every time. I have tried it for multiple users also I
tried using incognito and different browser but same behavior.

**Comment by cyraax** [comment]
I tested it on 12 computers and 3 different browsers.
I have the same problem.
The client sends a message, but we do not receive it on the message screen.
If I press F5 (Refresh), the message appears.

Version 4.0.1

**Comment by cyraax** [comment]
I use TRAEFIK here.
I had a problem solved   

**I added this environment variable**
environment:
      - ACTION_CABLE_URL=wss://xxx.domain.com/cable

labels:
        - traefik.enable=true
        - traefik.http.routers.chatwoot_app.rule=Host(`xxx.domain.com`)
        - traefik.http.routers.chatwoot_app.entrypoints=websecure
        - traefik.http.routers.chatwoot_app.tls.certresolver=letsencryptresolver
        - traefik.http.routers.chatwoot_app.service=chatwoot_app
        - traefik.http.services.chatwoot_app.loadbalancer.server.port=3000
        - traefik.http.services.chatwoot_app.loadbalancer.passhostheader=true
        **I have added these middleware settings below.**
        - traefik.http.routers.chatwoot_app.middlewares=chatwoot_ws_middleware
        - traefik.http.middlewares.chatwoot_ws_middleware.headers.customrequestheaders.X-Forwarded-Proto=https
        - traefik.http.middlewares.chatwoot_ws_middleware.headers.accesscontrolallowmethods=GET,OPTIONS,PUT
        - traefik.http.middlewares.chatwoot_ws_middleware.headers.accesscontrolallowheaders=DNT,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range
        - traefik.http.middlewares.chatwoot_ws_middleware.headers.accesscontrolexposeheaders=Content-Length,Content-Range
        - traefik.http.middlewares.chatwoot_ws_middleware.headers.accesscontrolalloworiginlist=*
        - traefik.http.middlewares.chatwoot_ws_middleware.headers.forcestsheader=true


I hope I helped

**Comment by arpeemidias** [comment]
@cyraax thanks! solved here.
